TASK-53132: Enable posting announcement challenge with emoji

ArfaouiMohamedAmine opened the pull request:
when an emoji is added to the description to announce a challenge, an sql error has been endorsed, it turns out that the description (named 'COMMENT' in the announecement table) is typed as varchar(2000), inorder to resolve this issue ; a type modification of utf8mb4 COLLATE utf8mb4_unicode_ci to support emojis has been added to the changelog. 
FYI: 
- locally the error is not reproduced .
- with this type modification it is still not reproduced for both mySQL qnd Postgres .

bourasom committed:
TASK-54163: Create FB meedsv2 and update projects versions/dependencies

bourasom committed:
TASK-54163: Create FB meedsv2 and update projects versions/dependencies

GouadriaHanen committed:
test build failure

GouadriaHanen committed:
test build failure

MayTekayaa committed:
Task-57295: Make clickable action label in Realizations table (#336)

Action labels in Realizations table are updated to be clickable

MayTekayaa committed:
Task-57295: Make clickable action label in Realizations table (#336)

Action labels in Realizations table are updated to be clickable

ArfaouiMohamedAmine committed:
TASK-55174: Fix Technical label in the description field when editing a newly rule (#343)

* Fix :Technical names in the description field when editing a newly Rule

* code refactor by using description()

* eliminating logs

* set the description to this.editedrule.description

ArfaouiMohamedAmine committed:
TASK-55174: Fix Technical label in the description field when editing a newly rule (#343)

* Fix :Technical names in the description field when editing a newly Rule

* code refactor by using description()

* eliminating logs

* set the description to this.editedrule.description

GouadriaHanen committed:
Task 50956: add search filter for challenges (#335)

GouadriaHanen committed:
Task 50956: add search filter for challenges (#335)

GouadriaHanen committed:
TASK-55183:fix displaying and recreating disabled domains (#339)

GouadriaHanen committed:
TASK-55183:fix displaying and recreating disabled domains (#339)

GouadriaHanen committed:
TASK-55183:fix displaying and recreating disabled domains (#339)

ArfaouiMohamedAmine committed:
trying to change the column COMMENT its type from varchar(2000) to NVARCHAR(2000) SET utf8mb4 COLLATE utf8mb4_unicode_ci to support special caracters

ArfaouiMohamedAmine committed:
Update gamification.db.changelog-1.0.0.xml

GouadriaHanen commented on services/src/main/resources/db/changelog/gamification.db.changelog-1.0.0.xml:
```diff
@@ -419,4 +419,10 @@ Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
         <addAutoIncrement columnDataType="BIGINT" columnName="CHALLENGE_MANAGER_ID" incrementBy="1" startWith="1"
                           tableName="CHALLENGE_MANAGER_RULE"/>
     </changeSet>
+    <changeSet id="1.0.0-37" author="exo-gamification" dbms="mysql">
```
we need to change the type for all DB type not just mysql

ArfaouiMohamedAmine replied:
and by not specifying the dbms value, is it going to target all the db types by default ? should I delete the dbms or repeat this changeset for all the db types ?

GouadriaHanen replied:
you need to specify the dbms value : ' mysql,mssql,h2,sybase,db2,hsqldb,postgresql" '

ArfaouiMohamedAmine committed:
Update gamification.db.changelog-1.0.0.xml

ArfaouiMohamedAmine committed:
Update gamification.db.changelog-1.0.0.xml

ahamdi committed:
Merge branch 'feature/meedsv2' into TASK-53132

ahamdi commented on services/src/main/resources/db/changelog/gamification.db.changelog-1.0.0.xml:
```diff
@@ -419,4 +419,10 @@ Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
         <addAutoIncrement columnDataType="BIGINT" columnName="CHALLENGE_MANAGER_ID" incrementBy="1" startWith="1"
                           tableName="CHALLENGE_MANAGER_RULE"/>
     </changeSet>
+    <changeSet id="1.0.0-37" author="exo-gamification" dbms="mysql,mssql,h2,sybase,db2,hsqldb,postgresql">
```
this changeset is specific to Mysql, do you reproduce the problem on Postgresql ?
Otherwise, we do support just Mysql and Postgresql, no need to mention other RDBMS

ArfaouiMohamedAmine replied:
roger that !
i'll verify Postgresql's influence on the problem 
I want to mention that locally the problem is not reproduced, it's reproduced on the acceptance .
the instance logs of acceptance are pointing out that it's an SQL column type issue 'COMMENT' column .

ahamdi reviewed (commented):
Please check my comment.

ahamdi reviewed (changes requested):
@ArfaouiMohamedAmine Please try to reproduce on local environment, you have to use the same collation for yopur local Mysql instance as the one used in Acceptance. 
We can not merge without knowing real cause. Could you take more time to investigate ?

ArfaouiMohamedAmine committed:
Update gamification.db.changelog-1.0.0.xml

the used dbms are mentionned after testing the proposed solution on them according to the instance logs.
I want to mention that the issue is not reporduced locally (acceptance bug) so the change that has been proposed is a way of prooving that it wont change the behaviour of the challenge announcement locally and on the acceptance . 
h2 dbms is kept because it's used by default .

ahamdi committed:
Merge branch 'feature/meedsv2' into TASK-53132

ahamdi commented:
@ArfaouiMohamedAmine Please update the description and the description with the title and description of what was done and why as explained in the PR best practices https://developer.exoplatform.org/#id-pull-requests-best-practices

ArfaouiMohamedAmine commented:
@ahamdi the requested modifications have been added .

ahamdi committed:
Merge branch 'feature/meedsv2' into TASK-53132

ArfaouiMohamedAmine committed:
Update gamification.db.changelog-1.0.0.xml

eliminating h2 dbms

ArfaouiMohamedAmine commented:
@ahamdi found out that locally the column type is not updated 
so this change we made will do the work .

ahamdi commented on services/src/main/resources/db/changelog/gamification.db.changelog-1.0.0.xml:
```diff
@@ -419,4 +419,10 @@ Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
         <addAutoIncrement columnDataType="BIGINT" columnName="CHALLENGE_MANAGER_ID" incrementBy="1" startWith="1"
                           tableName="CHALLENGE_MANAGER_RULE"/>
     </changeSet>
+    <changeSet id="1.0.0-37" author="exo-gamification" dbms="mysql,mssql">
```
```suggestion
    <changeSet id="1.0.0-37" author="exo-gamification" dbms="mysql">
```

ArfaouiMohamedAmine committed:
Update services/src/main/resources/db/changelog/gamification.db.changelog-1.0.0.xml

Co-authored-by: Ali HAMDI <ahamdi@exoplatform.com>

ArfaouiMohamedAmine commented:
when an emoji is added to the description to announce a challenge, an sql error has been endorsed, it turns out that the description (named 'COMMENT' in the announecement table) is typed as varchar(2000), inorder to resolve this issue ; a type modification of utf8mb4 COLLATE utf8mb4_unicode_ci to support emojis has been added to the changelog.
FYI:

locally the error is not reproduced .
with this type modification it is still not reproduced for both mySQL .
